File: Android/StartActivityForResult/app/src/main/java/com/example/startactivityforresult/MainActivity.java
package com.example.startactivityforresult;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    TextView textView;
    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView=(TextView) findViewById(R.id.txt_view);
        button=(Button) findViewById(R.id.b_button);

        button.setOnClickListener(this);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode==2)
        {
            if(resultCode==3) {
                String message = data.getStringExtra("MESSAGE");
                textView.setText(message);
            }
        }
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.b_button:
                Intent i=new Intent(MainActivity.this,MainActivity2.class);
                startActivityForResult(i,2);
        }

    }
}

File: Android/demo2/app/src/main/java/com/example/demo/MainActivity.kt
package com.example.demo

import android.content.pm.PackageManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.security.keystore.KeyGenParameterSpec
import android.security.keystore.KeyProperties
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKeys


class MainActivity : AppCompatActivity() {


    var advancedKeyAlias: String? = null;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        encryptToSharedPref("hello");
        fun getValueFromSharedPreferencesWith(key: String) = getEncryptedPreferences().getString(key, null);

    }

    init {
        val advancedSpec = KeyGenParameterSpec.Builder("your_master_key_name",
            KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT).apply {
            setBlockModes(KeyProperties.BLOCK_MODE_GCM)
            setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
            setKeySize(256)
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
                val hasStrongBox = packageManager.hasSystemFeature(PackageManager.FEATURE_STRONGBOX_KEYSTORE)
                if (hasStrongBox)
                    setIsStrongBoxBacked(true)
            }
        }.build()
         advancedKeyAlias = MasterKeys.getOrCreate(advancedSpec)
    }

    private fun encryptToSharedPref(data: String){
        getEncryptedPreferences().edit().putString("android_crypto", data).apply()
    }


    private fun getEncryptedPreferences() =
        EncryptedSharedPreferences.create("your_shared_preferences", advancedKeyAlias.toString(),
            this, EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM)



}